### PR TITLE
[buffer] Update Buffer test to allow out of memory errors

### PR DESCRIPTION
**Pull request by grgustaf**
Signed-off-by: Geoff Gustafson <geoff@linux.intel.com>

### CODE REVIEW

**Comment by jimmy-huang** [comment]
+1

File: tests/test-buffer.js [hunk 1]
```diff
@@ -9,14 +9,28 @@ var buff;
...
     assert(buff.length === lens[i][1],
-           "The length of Buffer(" + lens[i][0] + ") expected:" + lens[i][1] +
-           " got:" + buff.length);
+           "Creating buffer of length " + lens[i][0] + " expected:" +
+           lens[i][1] + " got:" + buff.length);
+}
+
+// thrown errors are allowed on these lengths
+lens = [[4294967295, 4294967295],
+        [4294967296, 4294967295]];
```
Comment (haoxli) [inline tests/test-buffer.js]: I think we'd better to use 1073741823(2^30-1) following Node.js 6.x [buffer.kMaxLength
](https://nodejs.org/dist/latest-v6.x/docs/api/buffer.html#buffer_buffer_kmaxlength), and it's expected to be a valid length for large memory(>1G) or throw error when devices have no enough available memory.

1073741824(2^30) should always throw error for out of memory.

Comment (grgustaf) [inline tests/test-buffer.js]: This wouldn't be bad, but it's kind of irrelevant for these small devices we're really targeting. They will always run out of RAM way before this "max length". I guess I can add it under a Linux ifdef so it doesn't add more code to the small device case.

**Comment by grgustaf** [comment]
@jimmy-huang there are non-trivial updates to review. But I've tested with all the buffer tests.

**Pull request merged**